### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by NicholasCF**
The helpers for the calendar

### CODE REVIEW

File: app/helpers/bookings_helper.rb [hunk 1]
```diff
@@ -1,2 +1,37 @@
 module BookingsHelper
+  def day_bookings(date)
+    times = retrieve_period(date)
+    sz = times.length
```
Comment (sreycodes) [inline app/helpers/bookings_helper.rb]: Improve the variable naming

File: app/helpers/bookings_helper.rb [hunk 2]
```diff
@@ -1,2 +1,37 @@
...
+    idx = 1;
+
+    time1 = times[idx-1]
```
Comment (sreycodes) [inline app/helpers/bookings_helper.rb]: why is it outside and inside the while loop

**Comment by sreycodes** [comment]
Overall the code for ```day_bookings``` seems rather verbose, un-understandable and not very ruby-ish. Try to make it better.

File: app/models/booking.rb [hunk 3]
```diff
@@ -34,32 +34,67 @@ class Booking < ApplicationRecord
...
+
+    if duplicates.any?
+      self.errors.add(:base, "Similar pending booking exists, please check")
+    end  
+  end
+
   def enough_items
-    used = Booking.where(item_id: item_id, status: "approved").where('? BETWEEN ? AND ?', self.start_time, start_time, end_time).size
-    available = self.item.quantity - used
+    return unless self.errors.blank?
```
Comment (Loop97) [inline app/models/booking.rb]: What's the purpose of this function enough_items? Is it to check if we have enough items left in inventory?

File: spec/factories/bookings.rb [hunk 4]
```diff
@@ -26,10 +26,13 @@
...
+    association :item, factory: :item
+
     status { 1 }
-    quantity {100}
+    quantity { 1 }
     description { "MyText" }
-    start_time { "2030-02-24 05:00:00" }
-    end_time { "2031-02-24 05:00:00" }
+    start_time { "2000-01-01 09:00:00" }
+    end_time { "2000-01-15 10:00:00" }
```
Comment (Loop97) [inline spec/factories/bookings.rb]: Should the start time and end time be a 24hours span or opening hours span? instead of two weeks?

**Review by Loop97 — APPROVED** [review]

File: app/models/booking.rb (hunk 3 above)
Comment (sreycodes) [reply app/models/booking.rb]: yes

File: app/models/booking.rb [hunk 5]
```diff
@@ -34,23 +34,38 @@ class Booking < ApplicationRecord
   validates :quantity, numericality:{greater_than: 0}
   validates :item_id, presence: true
   validates :user_id, presence: true
-  validates :start_time, uniqueness: { scope: [:item_id, :user_id, :end_time] }
```
Comment (sreycodes) [inline app/models/booking.rb]: Why was this validation removed?

**Pull request merged**